### ISSUE TITLE
Windows 10 (latest commit test) issue

### ISSUE THREAD

**Issue by ouje** [phaethon/scapy#37]
from scapy.all import *
  File "C:\Python34\lib\site-packages\scapy\all.py", line 16, in <module>
    from .arch import *
  File "C:\Python34\lib\site-packages\scapy\arch\__init__.py", line 88, in <module>
    from .windows import *
  File "C:\Python34\lib\site-packages\scapy\arch\windows\__init__.py", line 201, in <module>
    ifaces.load_from_powershell()
  File "C:\Python34\lib\site-packages\scapy\arch\windows\__init__.py", line 151, in load_from_powershell
    for i in get_windows_if_list():
  File "C:\Python34\lib\site-packages\scapy\arch\windows\__init__.py", line 91, in get_windows_if_list
    current_interface['name'] = value.decode('ascii')
UnicodeDecodeError: 'ascii' codec can't decode byte 0xa1 in position 1: ordinal not in range(128)

**Comment by phaethon** [comment]
What is the interface language for your Windows?

Please, open powershell, execute:
`Get-NetAdapter | select Name, ifIndex, InterfaceDescription, MacAddress`
and paste the output here.

**Comment by ouje** [comment]
Thanks for the idea! I changed the encoding format >
            current_interface['name'] = value.decode('ISO 8859-2')
        elif name == b'InterfaceIndex':
            current_interface['win_index'] = int(value)
        elif name == b'InterfaceDescription':
            current_interface['description'] = value.decode('ISO 8859-2')

and it works good for me. My windows uses latin.

**Comment by phaethon** [comment]
Not a general solution though. I need to get some ideas how to determine encoding in a general case. Posted a question on StackOverflow http://stackoverflow.com/questions/33936074/determine-default-windows-display-language-encoding-from-python

**Comment by phaethon** [comment]
Actually, problem seems much more complex than expected. Extensive googling and question on StackOverflow did not bring a result. Also, I cannot reproduce the issue. I am on Windows 10. I changed my network adapter to a name containing non-ascii characters, but the script still works with .decode('ascii') as non-ascii characters are translated to closest ascii equivalents. @ouje which Windows version are you on? Can you come up with an idea how can I determine from python script that I need to decode it from 8859-2? Or may be I just need a parameter in config - winencoding or smth like that.

**Comment by phaethon** [comment]
Actually I have found a solution. Please check with latest code from git.

**Comment by ouje** [comment]
I checked it and I can confirm the functionality with the latin encoding:

from function > get_windows_if_list()
[{'name': '__Sˇś Ethernet__', 'mac':  ## correct phrase should be " __Síť Ethernet__"
Process finished with exit code 0

**Comment by phaethon** [comment]
Well, at least it works without exception now. For somebody interested in fixing this - StackOverflow link above has detailed answer on powershell and string encodings.

**Comment by kartongsaft** [comment]
When is next release planned where this if fixed? I have the same issue on Windows 10.

**Comment by phaethon** [comment]
No issue fix currently in schedule. I don't have non-English Windows to test on. Patches are welcome. You use GitHub version or pip version?

**Comment by kartongsaft** [comment]
I'm currently using the version installed from pip.

**Comment by phaethon** [comment]
Do you get exception or names are just not correct?

**Comment by kartongsaft** [comment]
It's related to the name of my network adapters. If I change the name from Bluetooth-nätverksanslutning to Bluetooth-natverksanslutning, it starts working right away.

**Comment by phaethon** [comment]
Yes, that's why it is marked low priority as it has a workaround. But did you get exception or the names just looked wrong?
